src/librssguard/core/messagesproxymodel.cpp

```diff
@@ -100,6 +100,44 @@ bool MessagesProxyModel::lessThan(const QModelIndex& left, const QModelIndex& ri
   return false;
 }
 
+bool MessagesProxyModel::acceptMessage(Message currentMessage) const {
+  const QDate currentDate = QDate::currentDate();
```

Comment:
These two lines are only used in some switch cases, rearrange the method to only make those calls when needed.

Usually these methods which are used for repeated calling (especially in "proxy" and "source" models) must be optimized as much as possible. Imagine displaying 10 000 articles - which is not rare in RSS Guard - and calling `const QDateTime currentDateTime = QDateTime::currentDateTime();` uselessly 10 000 times. Also only call `currentDateTime` and then call `.date()` on it, it's bit faster.